test/verify.tests.js

```diff
@@ -67,6 +69,88 @@ describe('verify', function() {
     });
   });
 
+  describe('secret or token as callback', function () {
+      var token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJmb28iOiJiYXIiLCJpYXQiOjE0MzcwMTg1ODIsImV4cCI6MTQzNzAxODU5Mn0.3aR3vocmgRpG05rsI9MpR6z2T_BGtMQaPq2YR6QaroU';
+      var key = 'key';
+
+      var payload = { foo: 'bar', iat: 1437018582, exp: 1437018592 };
+      var options = {algorithms: ['HS256'], ignoreExpiration: true};
+
+      it('without callback', function (done) {
+          jwt.verify(token, key, options, function (err, p) {
+              assert.isNull(err);
+              assert.deepEqual(p, payload);
+              done();
+          });
+      });
+
+      it('simple callback', function (done) {
+          var keyFunc = function(header, callback) {
+              callback(undefined, key);
```

Comment:
I just saw that even when you are passing the `decodedToken` the tests pass. Can you assert here you get just the content of the header from the token?